libraries/WiFi/src/WiFiGeneric.cpp

```diff
@@ -1590,6 +1609,37 @@ int WiFiGenericClass::hostByName(const char* aHostname, IPAddress& aResult)
     return (uint32_t)aResult != 0;
 }
 
+/**
+ * Resolve the given hostname to an IP6 address.
+ * @param aHostname     Name to be resolved
+ * @param aResult       IPv6Address structure to store the returned IP address
+ * @return 1 if aHostname was successfully converted to an IP address,
+ *          else error code
+ */
+int WiFiGenericClass::hostByName6(const char* aHostname, ip_addr_t& aResult)
```

Comment:
this should accept `IPAddress` instead. We use Arduino APIs if we can. Also this function is not guarded if v6 is enabled.

Reply:
I will check this one, I think it comes from ESP8266